server/resourcemanager/testutil/test_resource.go

```diff
@@ -0,0 +1,114 @@
+package testutil
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/yaml"
+)
+
+type Operation string
+
+type ResourceTypeTest struct {
+	ResourceType      string
+	RegisterManagerFn func(*mux.Router) any
+	Prepare           func(operation Operation, bridge any)
+
+	SampleJSON string
+}
+
+type contentType struct {
+	name        string
+	contentType string
+	fromJSON    func(input string) string
+	toJSON      func(input string) string
+}
+
+type operationTester interface {
+	buildRequest(*testing.T, *httptest.Server, contentType, ResourceTypeTest) *http.Request
+	assertResponse(*testing.T, *http.Response, contentType, ResourceTypeTest)
+	name() Operation
+}
+
+var contentTypes = []contentType{
+	{
+		name:        "json",
+		contentType: "application/json",
+		fromJSON:    func(jsonSring string) string { return jsonSring },
+		toJSON:      func(jsonSring string) string { return jsonSring },
+	},
+
+	{
+		name:        "yaml",
+		contentType: "text/yaml",
+		fromJSON: func(jsonString string) string {
+			y, err := yaml.JSONToYAML([]byte(jsonString))
+			if err != nil {
+				panic(err)
+			}
+			return string(y)
+		},
+		toJSON: func(yamlString string) string {
+			j, err := yaml.YAMLToJSON([]byte(yamlString))
+			if err != nil {
+				panic(err)
+			}
+			return string(j)
+		},
+	},
+}
+
+func TestResourceType(t *testing.T, rt ResourceTypeTest) {
+	t.Helper()
+
+	t.Run(rt.ResourceType, func(t *testing.T) {
+		rt := rt
```

Comment:
I'm almost sure you don't need this line because `rt` doesn`t change. We only need to do this thing when we are working with parallel tests inside a loop. Just like in did on line 73.

Reply:
yeap, I forgot to `t.Parallel()` here

Reply:
You're right. I'll clean this up and check if i didn't repeat anything